keanu-project/src/main/java/io/improbable/keanu/vertices/bool/BooleanVertex.java

```diff
@@ -0,0 +1,89 @@
+package io.improbable.keanu.vertices.bool;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.vertices.Observable;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.operators.binary.BooleanBinaryOpVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.operators.multiple.AndMultipleVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.operators.multiple.OrMultipleVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.operators.unary.BooleanUnaryOpVertex;
+import io.improbable.keanu.vertices.update.ValueUpdater;
+
+public abstract class BooleanVertex extends Vertex<BooleanTensor> {
+
+    public BooleanVertex(ValueUpdater<BooleanTensor> valueUpdater, Observable<BooleanTensor> observation) {
+        super(valueUpdater, observation);
+    }
+
+    @SafeVarargs
```

Comment:
@gordoncaleb I think there's no need for `@SafeVarargs` here?

Reply:
Yeah I think this is just to avoid the warning.